command/resource/client/grpc-config.go

```diff
@@ -4,40 +4,118 @@
 package client
 
 import (
+	"fmt"
 	"os"
+	"strconv"
+	"strings"
 )
 
 const (
 	// GRPCAddrEnvName defines an environment variable name which sets the gRPC
 	// server address for the consul CLI.
 	GRPCAddrEnvName = "CONSUL_GRPC_ADDR"
+
+	// GRPCTLSEnvName defines an environment variable name which sets the gRPC
+	// communication mode. Default is false in plaintext mode.
+	GRPCTLSEnvName = "CONSUL_GRPC_TLS"
+
+	// GRPCSSLVerifyEnvName defines an environment variable name which sets
+	// whether to disable certificate checking.
+	GRPCSSLVerifyEnvName = "CONSUL_GRPC_SSL_VERIFY"
+
+	// GRPCCAFileEnvName defines an environment variable name which sets the
+	// CA file to use for talking to Consul gRPC over TLS.
+	GRPCCAFileEnvName = "CONSUL_GRPC_CACERT"
+
+	// GRPCCAPathEnvName defines an environment variable name which sets the
+	// path to a directory of CA certs to use for talking to Consul gRPC over TLS.
+	GRPCCAPathEnvName = "CONSUL_GRPC_CAPATH"
 )
 
 type GRPCConfig struct {
+	// Address is the optional address of the Consul server in format of host:port.
+	// It doesn't include schema
 	Address string
+
+	// GRPCTLS is the optional boolean flag to determine the communication protocol
+	GRPCTLS bool
+
+	// GRPCSSLVerifyEnvName is the optional boolean flag to disable certificate checking.
+	// Set to false only if you want to skip server verification
+	GRPCSSLVerifyEnvName bool
```

Comment:
nit: this should probably be named `GRPCTLSVerify` as this does not represent the name of an env var any longer.